bazel/envoy_library.bzl

```diff
@@ -115,27 +115,10 @@ EXTENSION_STATUS_VALUES = [
 
 def envoy_cc_extension(
         name,
-        security_posture,
-        category = None,
-        # Only set this for internal, undocumented extensions.
-        undocumented = False,
-        status = "stable",
         tags = [],
         extra_visibility = [],
         visibility = EXTENSION_CONFIG_VISIBILITY,
         **kwargs):
-    if not category:
-        fail("Category not set for %s" % name)
```

Comment:
figure out what to do about this

when this macro is called there is not a way of knowing (i think) what extension its being called so we cant do a lookup from the bazel extension conf

we could add to all of the extensions their `extension_name` - ie `envoy.access_loggers.file` - and then we *could* look up the key in the config data to check the extension meets the tests that are removed

Reply:
adding the `extension_name` is in some ways redundant because the `extensions_build_config.bzl` already has a `source` field 

*but* it would make life easier for devs - ie when reading the source bazel files, the extension name is explicit and you can quickly check/copypasta-search in `extensions_build_config.bzl`

Reply:
the other option is to  move these validation checks to `generate_extension_db` which already has some validation

Reply:
if we dont move the validation to `generate_extension_db` then im wondering whether we should consolidate the checks that happen there to here